data/content_scripts/posting_process/privly_app.js

```diff
@@ -0,0 +1,58 @@
+/**
+ * @fileOverview Script acting as a "middleman" to allow communication between
+ * the extension and privly application.
+ */
+
+/**
+ * @namespace Posting Process
+ * If postingProcess namespace is not initialized, initialize it
+ */
+var postingProcess;
+if (postingProcess === undefined) {
+  postingProcess = {};
+}
+
+/**
+ * Handles the receipt of initial content from the extension. This content is then 
+ * sent to the privly application
+ * 
+ * @param {Object} data Message sent by the extension. This includes the initial content 
+ *                      and the message secret(used to establish communication between the
+ *                      content scripts and privly-applications).
+ */
+postingProcess.initialContentHandler = function(data) {
+  var message = JSON.stringify({
+                  secret: data.secret,
+                  handler: "initialContent",
+                  initialContent: data.initialContent,
+                });
+  window.postMessage(message, "*");
+}
+
+window.addEventListener("PrivlyMessageEvent", function(e) {
```

Comment:
It is better to define the function separately for documentation and testing purposes.